### PR TITLE
Add providers

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/yii-web/issues/259

### CODE REVIEW

**Comment by terabytesoftw** [comment]
I can add add MiddlewareProvider, but you have to define:

- [ ] FileRotatorProvider
- [ ] FileTargetProvider
- [ ] LoggerProvider
- [ ] RouteCollectorProvider
- [ ] UrlGeneratorProvider
- [ ] UrlMatcherProvider

File: README.md [hunk 1]
```diff
@@ -17,79 +17,38 @@ This [Yii Framework] extension allows easy creation of web applications.
...
-
-### Directory Structure
-
-```
-build/               internally used build tools
-docs/                documentation
-framework/           core framework code
-tests/               tests of the core framework code
+```php
+composer require yiisoft/yii-wev
```
Comment (samdark) [inline README.md:28]:
```suggestion
composer require yiisoft/yii-web
```

Comment (terabytesoftw) [reply README.md]: done.

File: composer.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "ext-json": "*",
         "ext-dom": "*",
         "alexkart/curl-builder": "^1.0",
+        "nyholm/psr7": "^1.0",
```
Comment (samdark) [inline composer.json:24]: That ties us to using one concrete PSR-7 implementation.

File: src/Provider/ThrowableRendererProvider.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+use Psr\Container\ContainerInterface;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\Support\ServiceProvider;
+use Yiisoft\Yii\Web\ErrorHandler\HtmlRenderer;
+use Yiisoft\Yii\Web\ErrorHandler\ThrowableRendererInterface;
+
+final class ThrowableRendererProvider extends ServiceProvider
+{
+    private array $templates;
```
Comment (samdark) [inline src/Provider/ThrowableRendererProvider.php:15]: Doesn't seem to be used anywhere.

File: tests/Provider/HtmlRendererProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+namespace Yiisoft\Yii\Web\Tests\Provider;
+
+use Yiisoft\Yii\Web\ErrorHandler\HtmlRenderer;
+use Yiisoft\Yii\Web\Tests\TestCase;
+
+final class HtmlRendererProviderTest extends TestCase
+{
+    public function testProviderConfig(): void
+    {
+        $this->assertNotEmpty($this->container->get(HtmlRenderer::class));
```
Comment (samdark) [inline tests/Provider/HtmlRendererProviderTest.php:14]:
```suggestion
        $this->assertInstanceOf(HtmlRenderer::class, $this->container->get(HtmlRenderer::class));
```

File: tests/Provider/Psr17ProviderTest.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\UploadedFileFactoryInterface;
+use Psr\Http\Message\UriFactoryInterface;
+use Yiisoft\Yii\Web\Tests\TestCase;
+
+final class Psr17ProviderTest extends TestCase
+{
+    public function testProviderConfig(): void
+    {
+        $this->assertNotEmpty($this->container->get(RequestFactoryInterface::class));
```
Comment (samdark) [inline tests/Provider/Psr17ProviderTest.php:19]: It's better to use `assertInstanceOf` for these.

File: tests/Provider/SessionProviderTest.php [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+namespace Yiisoft\Yii\Web\Tests\Provider;
+
+use Yiisoft\Yii\Web\Session\SessionInterface;
+use Yiisoft\Yii\Web\Tests\TestCase;
+
+final class SessionProviderTest extends TestCase
+{
+    public function testProviderConfig(): void
+    {
+        $this->assertNotEmpty($this->container->get(SessionInterface::class));
```
Comment (samdark) [inline tests/Provider/SessionProviderTest.php:14]: It's better to use `assertInstanceOf` for these.

File: tests/Provider/ThrowableRendererProviderTest.php [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+namespace Yiisoft\Yii\Web\Tests\Provider;
+
+use Yiisoft\Yii\Web\ErrorHandler\ThrowableRendererInterface;
+use Yiisoft\Yii\Web\Tests\TestCase;
+
+final class ThrowableRendererProviderTest extends TestCase
+{
+    public function testProviderConfig(): void
+    {
+        $this->assertNotEmpty($this->container->get(ThrowableRendererInterface::class));
```
Comment (samdark) [inline tests/Provider/ThrowableRendererProviderTest.php:14]: It's better to use `assertInstanceOf` for these.

**Review by samdark — CHANGES_REQUESTED** [review]

File: tests/Provider/ThrowableRendererProviderTest.php (hunk 7 above)
Comment (terabytesoftw) [reply tests/Provider/ThrowableRendererProviderTest.php]: done.

File: tests/Provider/SessionProviderTest.php (hunk 6 above)
Comment (terabytesoftw) [reply tests/Provider/SessionProviderTest.php]: done.

File: tests/Provider/Psr17ProviderTest.php (hunk 5 above)
Comment (terabytesoftw) [reply tests/Provider/Psr17ProviderTest.php]: done.

File: tests/Provider/HtmlRendererProviderTest.php (hunk 4 above)
Comment (terabytesoftw) [reply tests/Provider/HtmlRendererProviderTest.php]: done.

**Comment by samdark** [comment]
> I can add add MiddlewareProvider, but you have to define:

I think all these should be in the yiisoft/app. Doesn't make sense to have these at yii-web level.

File: src/Provider/ThrowableRendererProvider.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Provider/ThrowableRendererProvider.php]: done.

**Comment by samdark** [comment]
Decided to move all these to yiisoft/app instead.

**Pull request closed without merging**